File: 设计模式/设计模式/命令模式/单品/DanPinSetTimeCommand.m
//
//  DanPinSettimeCommand.m
//  设计模式
//
//  Created by zachary spark on 2019/5/13.
//  Copyright © 2019 3g. All rights reserved.
//

#import "DanPinSetTimeCommand.h"

@interface DanPinSetTimeCommand()
@property(nonatomic,strong)DanPinRealization *DanPin;
@end

@implementation DanPinSetTimeCommand

- (instancetype)init:(DanPinRealization *)DanPin{
    
    self = [super init];
    if(self){
        self.DanPin = DanPin;
    }
    return self;
    
}


-(void)execute{
    [self.DanPin ProductSetTime:self.interval];
}

@end

File: 设计模式-Swift/设计模式-Swift/命令模式/基本实现/DanPinSetTimeCommand.swift
//
//  DanPinSetTimeCommand.swift
//  设计模式-Swift
//
//  Created by zachary spark on 2019/5/14.
//  Copyright © 2019 3g. All rights reserved.
//

import UIKit

class DanPinSetTimeCommand: DanPinCommandProtocol {

    var DanPin:DanPinRealization
    
    var Interval:TimeInterval = 10  //默认10秒
    
    init(_ DanPin:DanPinRealization) {
        self.DanPin = DanPin
    }
    
    func execute() {
        self.DanPin.ProductSetTime(self.Interval as AnyObject)
    }
    
}
